### PR TITLE
[quadtree] Add port

### PR DESCRIPTION
**Pull request by meyerzinn**
Adds a simple, header-only [Quadtree](https://github.com/pvigier/Quadtree/) library, which is useful for game development.

* Which triplets are supported/not supported? Have you updated the CI baseline?
Tested to work on x64-linux (using WSL2) and x64-mingw, so I would venture a guess to say it works on most platforms. I did not touch the CI baseline but if anyone thinks it's necessary then I'd be happy to learn how.

### CODE REVIEW

File: ports/quadtree/CONTROL [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Source: quadtree
+Version: 1.0.0
```
Comment (NancyLi1013) [inline ports/quadtree/CONTROL:2]: It seems that there is no released version tagged as 1.0.0.
In this case, we might use the date `2020-04-13` as the version.
Please refer to this [doc](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/control-files.md#version) to learn more about this.

File: ports/quadtree/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO pvigier/Quadtree
+    REF fec2e1a011f656353ba12c58cefe33482e1a27b5
+    SHA512 8291afb29b9ff5714125e411122f6f10e1c2e4c2109384e25afa834548dc1f63c90b2d09c7cfed213ad69a8c54cf32fc598783b3ff0882bc86562594895d5c35
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    DISABLE_PARALLEL_CONFIGURE
```
Comment (NancyLi1013) [inline ports/quadtree/portfile.cmake:11]: Why do we need to add `DISABLE_PARALLEL_CONFIGURE` here?

File: ports/quadtree/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    OPTIONS
+        -DCODE_COVERAGE=OFF
+        -DBUILD_BENCHMARKS=OFF
+        -DBUILD_TESTING=OFF
+        -DBUILD_EXAMPLES=OFF
+)
+
+file(INSTALL ${SOURCE_PATH}/include/ DESTINATION ${CURRENT_PACKAGES_DIR}/include/quadtree)
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/quadtree/copyright)
```
Comment (NancyLi1013) [inline ports/quadtree/portfile.cmake:20]: Could you update the way to handle copyright as
`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${port} RENAME copyright) `?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/quadtree/portfile.cmake (hunk 2 above)
Comment (20zinnm) [reply ports/quadtree/portfile.cmake:11]: Not sure we do, I just copied the template from a different library. I'll take it out and see if the tests pass.

File: ports/quadtree/portfile.cmake (hunk 3 above)
Comment (20zinnm) [reply ports/quadtree/portfile.cmake]: `${port}` did not work for me so I replaced it with `quadtree`. Let me know if I did something wrong.

Comment (NancyLi1013) [reply ports/quadtree/portfile.cmake]: Sorry for my mistake, It should be ${PORT}.  Could you please try it again?

File: ports/quadtree/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    SOURCE_PATH ${SOURCE_PATH}
+    OPTIONS
+        -DCODE_COVERAGE=OFF
+        -DBUILD_BENCHMARKS=OFF
+        -DBUILD_TESTING=OFF
+        -DBUILD_EXAMPLES=OFF
+)
+
+file(INSTALL ${SOURCE_PATH}/include/ DESTINATION ${CURRENT_PACKAGES_DIR}/include/quadtree)
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/quadtree RENAME copyright)
```
Comment (NancyLi1013) [inline ports/quadtree/portfile.cmake:19]: Could you please update the way to handle copyright as
`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Pull request merged**